trafficlight/tests/fallback_key_test.py

```
@@ -0,0 +1,85 @@
+        await bob.accept_invite()
+        # disable sync for alice, so she can't upload more device keys
+        await alice_proxy.disable_endpoint("/_matrix/client/r0/sync")
+        for i in range(60):
+            await login_and_send_message_in_room(
+                server, bob, room_id, f"Hello world {i + 1}!"
+            )
+        await alice_proxy.enable_endpoint("/_matrix/client/r0/sync")
+        # last message would have exhausted the OTKs,
+        # so we should have fallen back to the fallback key
+        await alice.verify_message_in_timeline("Hello world 60!")
+
+
+async def login_and_send_message_in_room(
+    server: HomeServer, user: MatrixClient, room_id: str, message: str
+) -> None:
+    # need to install python-olm and pip install "matrix-nio[e2e]
+    logging.info(
+        f"trying to login as @{user.localpart}:{server.server_name} with password ${user.password} and send a message in #{room_id}..."
+    )
+    user_id = f"@{user.localpart}:{server.server_name}"
+    client = AsyncClient(
+        server.cs_api,
+        user_id,
+        config=AsyncClientConfig(
+            encryption_enabled=True, store=store.SqliteMemoryStore
+        ),
+    )
+
+    # This method will be called after each sync
+    async def handle_sync(_: None) -> None:
+        # Send the message after initial sync
+        await client.room_send(
+            room_id,
+            message_type="m.room.message",
+            content={"msgtype": "m.text", "body": message},
+            ignore_unverified_devices=True,
+        )
+        # Stop syncing
+        task.cancel()
+
+    try:
+        client.add_response_callback(handle_sync, SyncResponse)
+        await client.login(user.password)
+        # sync_forever must be used for encryption to work
+        task = asyncio.create_task(client.sync_forever(timeout=30000))
+        await task
+    except Exception as e:
+        logging.exception(str(e))
```

What exception are we catching, logging and dropping here?

If we do it like this, the exception handling of the test suite as a whole won't mark the test as failed - are we using this to hide an exception that isn't fatal to the test?

A short note on what we're catching/why would be good if we do need to hide exceptions that aren't failures.